### ISSUE TITLE
[Java 21] TestLimitModules failure with Java 21

### ISSUE THREAD

**Issue by SarikaSinha** [eclipse-jdt/eclipse.jdt.debug#298]
https://download.eclipse.org/eclipse/downloads/drops4/Y20230827-1000/testresults/html/org.eclipse.jdt.debug.tests_ep429Y-unit-cen64-gtk3-java21_linux.gtk.x86_64_21.html

org.eclipse.jdt.debug.tests.core.ModuleOptionsTests | testLimitModules1 | Failure | Unknown set of default modules java.se,jdk.accessibility,jdk.attach,jdk.compiler,jdk.dynalink,jdk.httpserver,jdk.incubator.vector,jdk.jartool,jdk.javadoc,jdk.jconsole,jdk.jdi,jdk.jfr,jdk.jshell,jdk.jsobject,jdk.management,jdk.management.jfr,jdk.net,jdk.nio.mapmode,jdk.sctp,jdk.security.auth,jdk.security.jgss,jdk.unsupported,jdk.unsupported.desktop,jdk.xml.domjunit.framework.AssertionFailedError: Unknown set of default modules java.se,jdk.accessibility,jdk.attach,jdk.compiler,jdk.dynalink,jdk.httpserver,jdk.incubator.vector,jdk.jartool,jdk.javadoc,jdk.jconsole,jdk.jdi,jdk.jfr,jdk.jshell,jdk.jsobject,jdk.management,jdk.management.jfr,jdk.net,jdk.nio.mapmode,jdk.sctp,jdk.security.auth,jdk.security.jgss,jdk.unsupported,jdk.unsupported.desktop,jdk.xml.domat junit.framework.Assert.fail(Assert.java:57)at junit.framework.TestCase.fail(TestCase.java:223)at org.eclipse.jdt.debug.tests.core.ModuleOptionsTests.testLimitModules1(ModuleOptionsTests.java:295)at java.base/java.lang.reflect.Method.invoke(Method.java:580)at junit.framework.TestCase.runTest(TestCase.java:177)at junit.framework.TestCase.runBare(TestCase.java:142)at org.eclipse.jdt.debug.tests.AbstractDebugTest.runBare(AbstractDebugTest.java:2678)at junit.framework.TestResult$1.protect(TestResult.java:122)at junit.framework.TestResult.runProtected(TestResult.java:142)at junit.framework.TestResult.run(TestResult.java:125)at junit.framework.TestCase.run(TestCase.java:130)at junit.framework.TestSuite.runTest(TestSuite.java:241)at junit.framework.TestSuite.run(TestSuite.java:236)at junit.framework.TestSuite.runTest(TestSuite.java:241)at org.eclipse.jdt.debug.tests.DebugSuite$1.run(DebugSuite.java:61)at java.base/java.lang.Thread.run(Thread.java:1583)
-- | -- | -- | --

**Comment by SarikaSinha** [comment]
@mpalat

**Comment by SarikaSinha** [comment]
https://download.eclipse.org/eclipse/downloads/drops4/Y20230830-1000/testResults.php
Test doesn't fail any more.
